src/lib.rs

```diff
@@ -479,7 +479,7 @@ fn process_join(from: &Arc<Session>, room_id: RoomId, user_id: UserId, subscribe
         switchboard.join_publisher(Arc::clone(from), user_id.clone(), room_id.clone());
         notify_except(&notification, &user_id, switchboard.publishers_occupying(&room_id));
     } else {
-        switchboard.join_subscriber(Arc::clone(from), user_id.clone(), room_id.clone());
+        switchboard.join_subscriber(Arc::clone(from), user_id, room_id);
```

Comment:
If found this one with `cargo clippy`.
`user_id` and `room_id` are owned String from the `MessageKind::Join` message (serde DeserializeOwned) that are already moved to `process_join`. Those two String are not used after this, so no need to clone, `join_subscriber` can own them.

Note that in `process_block`, the `whom` String is not cloned, the String is moved to `switchboard.establish_block` without issue.